File: app/src/main/java/com/uni/phamduy/massagefinder/module/place/Ticket.java

package com.uni.phamduy.massagefinder.module.place;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Ticket {

    @SerializedName("id")
    @Expose
    private String id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("period")
    @Expose
    private Integer period;
    @SerializedName("price")
    @Expose
    private Integer price;
    @SerializedName("priceWithCode")
    @Expose
    private Integer priceWithCode;
    @SerializedName("description")
    @Expose
    private String description;
    @SerializedName("tip")
    @Expose
    private Integer tip;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getPeriod() {
        return period;
    }

    public void setPeriod(Integer period) {
        this.period = period;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }

    public Integer getPriceWithCode() {
        return priceWithCode;
    }

    public void setPriceWithCode(Integer priceWithCode) {
        this.priceWithCode = priceWithCode;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getTip() {
        return tip;
    }

    public void setTip(Integer tip) {
        this.tip = tip;
    }

}

File: app/src/main/java/com/uni/phamduy/massagefinder/module/place/Ticket.kt
package com.uni.phamduy.massagefinder.module.place

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Ticket() : Parcelable {

    @SerializedName("id")
    @Expose
    var id: String? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("period")
    @Expose
    var period: Int? = null
    @SerializedName("price")
    @Expose
    var price: Int? = null
    @SerializedName("priceWithCode")
    @Expose
    var priceWithCode: Int? = null
    @SerializedName("description")
    @Expose
    var description: String? = null
    @SerializedName("tip")
    @Expose
    var tip: Int? = null

    constructor(parcel: Parcel) : this() {
        id = parcel.readString()
        name = parcel.readString()
        period = parcel.readValue(Int::class.java.classLoader) as? Int
        price = parcel.readValue(Int::class.java.classLoader) as? Int
        priceWithCode = parcel.readValue(Int::class.java.classLoader) as? Int
        description = parcel.readString()
        tip = parcel.readValue(Int::class.java.classLoader) as? Int
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(id)
        parcel.writeString(name)
        parcel.writeValue(period)
        parcel.writeValue(price)
        parcel.writeValue(priceWithCode)
        parcel.writeString(description)
        parcel.writeValue(tip)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Ticket> {
        override fun createFromParcel(parcel: Parcel): Ticket {
            return Ticket(parcel)
        }

        override fun newArray(size: Int): Array<Ticket?> {
            return arrayOfNulls(size)
        }
    }

}
